src/main/java/org/spongepowered/common/mixin/core/block/BlockFenceMixin.java

```diff
@@ -37,14 +39,27 @@
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.common.data.ImmutableDataCachingUtil;
 import org.spongepowered.common.data.manipulator.immutable.block.ImmutableSpongeConnectedDirectionData;
+import org.spongepowered.common.util.DirectionalBlockUtils;
 
 import java.util.HashSet;
+import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 
 @Mixin(BlockFence.class)
 public abstract class BlockFenceMixin extends BlockMixin {
 
+    private static final Map<Direction, PropertyBool> DIRECTION_TO_PROPERTY_MAPPING;
+
+    static {
+        ImmutableMap.Builder<Direction, PropertyBool> directionToPropertyMappingBuilder = ImmutableMap.builder();
```

Comment:
This really should not be statically initialized, I'd rather a constructor injection to initialize the field.

Reply:
So you suggest using a non-static field? But this field is not an object property in any way in terms of OOP, purely supporting information. Or did I misunderstand you?

Reply:
If it's not static to the Block itself, and it's something as a translation, then there's a [TypeProvider](https://github.com/SpongePowered/SpongeCommon/blob/stable-7/src/main/java/org/spongepowered/common/registry/TypeProvider.java) bit of the registry implementation, and there's also [Constants that can contain these type mappings](https://github.com/SpongePowered/SpongeCommon/blob/a865595b36ab95a876c8ebd46b464b23a73b09aa/src/main/java/org/spongepowered/common/util/Constants.java#L1097-L1105) where you can make some better static constants like this rather than in the mixin itself.